### ISSUE TITLE
Converting Float to Double

### ISSUE THREAD

**Issue by vsendhil** [dakrone/cheshire#97]
I ran into an issue while generating json from Float. It converts to Double. I think the problem is here 
https://github.com/dakrone/cheshire/blob/5.5.0/src/cheshire/generate.clj#L37
So here a Float is converted to Double. It should be `(float ...)`.

**Comment by briprowe** [comment]
When serializing a float w/value `0.01` for example, it's printed as `0.009999999776482582`.

**Comment by gfredericks** [comment]
Seems like a reasonable analysis to me. I could merge a PR with that change and a test.

**Comment by dakrone** [comment]
Sounds totally reasonable

**Comment by cemerick** [comment]
Hit this this morning. A workaround until a fix is merged is:

```clojure
(extend-protocol cheshire.generate/JSONable
  Float
  (to-json [x ^com.fasterxml.jackson.core.JsonGenerator generator]
    (.writeNumber generator x)))
```

**Comment by dakrone** [comment]
Sorry this took so long, I pushed a fix and released 5.6.3 with it.
